### DISCUSSION TITLE
Why is there no way to alert when a good bolus time is? We need a (BG - IOB * correction factor > threshhold) alarm or endpoint

### DISCUSSION THREAD

**Discussion by gth001** [NightscoutFoundation/xDrip#2974] [Ideas]
My insulin pump synchronizes insulin on board with Night Scout and X-Drip.

And so a feature I've desperately wanted for years is some notification for when my insulin really isn't going to cover how high my blood sugar is. And no, this is NOT a high glucose alarm. If I already bolused for the high glucose, there's no point in an alarm.

I'm not a computer science guru, so I've always hit frustrating dead ends with this project.

But if you have any ideas at all to get me closer to this absolute dream that would improve my diabetic life so much and probably extend the lifetimes of millions of people if you implemented it in some manner, you would be an angel. I hope someone out there can appreciate the truth of this, and that my desperate plea doesn't just get washed away with the waves. I'm willing to do coding work. I'm willing to learn about technical things. I'm willing to pay for a Zapier account. My issue is it is not as easy as you think it is and I am so lost with how to implement this. Anything you can give, no matter how "i dunno maybe try x" it is, it gratefully appreciated.

To be succint: When bloodGlucose - nightscoutIOB * correctionFactor > 25 mg/dL I want a phone notification, so that I know when to give myseld 0.5 units from omnipod or pen.

**Comment by gth001** [comment]
I got a proof of concept working 

It could use some tweaks but works great for me :) 

https://taskernet.com/shares/?user=AS35m8lLfxS4u8saOvKItBRvbxVIhJ1WvckxQ5Ujc4%2FaXad38%2BAG%2B%2BJ%2BGWmuYp0djggKpA%3D%3D&id=Profile%3AxDrip+GLU+Var+%26+Carb+Calc

![image](https://github.com/NightscoutFoundation/xDrip/assets/10224533/fc3613dd-779a-47fe-b401-883707f538de)
![image](https://github.com/NightscoutFoundation/xDrip/assets/10224533/89b8c358-e49d-475a-9356-81cafc998380)

**Comment by Navid200** [comment]
Nightscout runs on a server.  xDrip runs on a mobile.
Nightscout has all the data xDrip has.
Have you submitted this request to Nightscout?
